docs/rfcs/001-iiif-sandbox.md

```diff
@@ -0,0 +1,174 @@
+ - POSTing the manifest to some configured endpoint that will accept it and host it
+
+The GitHub route is great but is subject to caching delays which quickly get annoying. Plus you need a GitHub account.
+The local web server is great for developers but an extra overhead for casual users or a training context.
+The last of these is one of the integration mechanisms manifest editor will support; institutions can implement their own endpoints to integrate with their own CMS, storage, or whatever.
+
+This RFC proposes a variant of the third that is always available to all users of the manifest editor, and is used for previewing - even if you have the first and/or second mechanisms above, you might not want to have to commit, or push to storage, every time you want to check on your work in the editor.
+
+It's a service that **any client, anywhere** can POST a IIIF manifest to, so that the manifest is available on the public web, over https, with the necessary CORS headers.
+
+## Background discussions
+
+ - https://github.com/digirati-co-uk/iiif-manifest-editor/issues/1
+ - https://github.com/digirati-co-uk/iiif-manifest-editor/discussions/3
+
+## Proposal
+
+`iiif-sandbox.digirati.io`
+
+The Manifest Editor has a built-in integration with this service.
+
+The exact technical implementation depends on platform. It is likely to be an AWS lambda + public (read) S3 bucket, or a Cloudflare worker + R2 storage.
+
+Any client (not just the manifest editor) can POST a IIIF Manifest payload (i.e., a JSON body) to the `/store` endpoint:
+
+```
+POST /store HTTP/1.1
+Host: iiif-sandbox.digirati.io
+Content-Type: application/json
+
+{
+  "id": <original-id>,
+  // rest of valid IIIF manifest
+}
+```
+
+There is no access control of any kind on this request, anyone can post.
+
+The request is handled by a worker/lambda or whatever serverless equivalent. This is written in JavaScript to take advantage of the Vault component, below.
+The worker will mint two identifiers, and to avoid any need for DB storage, these two identifiers are 2-way encrypted (not hashes) - you can obtain either of them from the other, if in possession of a secret. The worker is in possession of this secret but clients are not.
+
+
+The worker:
+
+1. Validate that the content is below the configured size threshold
+1. Validates that the content is valid JSON (parse it into a JS object)
+2. Loads the object into an instance of Vault (https://iiif-canvas-panel.netlify.app/docs/components/vault) - if it succeeds, it's a valid manifest.
+3. If either 1, 2 or 3 fail, return a 400 Bad Request, with a JSON body `{ "error": "Bad Request" }` (we can expand on that later)
+4. If it was successfully loaded into vault, mint the *manifest* identifier. This is like a Google doc link - an unguessable long string.
+5. Replace the `id` of the manifest in the uploaded JSON with `https://iiif-sandbox.digirati.io/p3/<manifest-identifier>.json`. The /p3/ path reserves a "namespace" for future versions.
```

Comment:
We can support p2 for legacy viewers - although it is lossy

Reply:
Maybe, instead of POST to /store, you can POST to /store/p2, or /store/p3, or both.

And because vault, you can post either to either and it will convert. But we don't unnecessarily fill bucket with unneeded p2s.